packages/backend/src/routers/apiV1Router.ts

```diff
@@ -135,15 +135,16 @@ apiV1Router.get(
     }),
 );
 
-apiV1Router.get(
-    lightdashConfig.auth.google.callbackPath,
+apiV1Router.get(lightdashConfig.auth.google.callbackPath, (req, res, next) => {
+    const { returnTo } = req.session.oauth || {};
+
     passport.authenticate('google', {
         failureRedirect: '/api/v1/oauth/failure',
-        successRedirect: '/api/v1/oauth/success',
+        successRedirect: `/api/v1/oauth/success?returnTo=${returnTo}`,
```

Comment:
I'm passing the /success callback a query parameter, instead of relying on the session (which is overwritten)